test/parallel/test-child-process-execfile.js

```diff
@@ -99,3 +100,23 @@ const execOpts = { encoding: 'utf8', shell: true };
   });
   execFile(process.execPath, [fixture, 0], { signal }, callback);
 }
+
+// Verify the execFile() stdout is the same as execFileSync().
+{
+  const file = 'echo';
+  const args = ['foo', 'bar'];
+
+  // Test with and without `{ shell: true }`
+  [
+    // Skipping shell-less test on Windows because its echo command is a shell built-in command.
+    ...(common.isWindows ? [] : [null]),
+    { shell: true },
+  ].forEach((options) => {
+    const execFileSyncStdout = execFileSync(file, args, options);
+    assert.deepStrictEqual(execFileSyncStdout, Buffer.from(`foo bar${os.EOL}`));
+
+    execFile(file, args, options, common.mustCall((_, stdout) => {
+      assert.deepStrictEqual(Buffer.from(stdout), execFileSyncStdout);
```

Comment:
I'd insist that we should use the `encoding` option here, working with strings is much easier and produce better error messages – instead of an obscure list of numbers that don't match, it's much easier to compare two strings to understand what's failing imho.
```suggestion
    ...(common.isWindows ? [] : [{ encoding: 'utf8' }]),
    { shell: true, encoding: 'utf8' },
  ].forEach((options) => {
    const execFileSyncStdout = execFileSync(file, args, options);
    assert.strictEqual(execFileSyncStdout, `foo bar${os.EOL}`);

    execFile(file, args, options, common.mustCall((_, stdout) => {
      assert.strictEqual(stdout, execFileSyncStdout);
```